src/Compilers/CSharp/Test/Semantic/Semantics/SimpleLambdaParametersWithModifiersTests.cs

```diff
@@ -0,0 +1,1339 @@
+            delegate void D({{(delegateIsParams ? "params" : "")}} int[] x);
+
+            class C
+            {
+                void M()
+                {
+                    D d = (params x) =>
+                    {
+                    };
+                }
+            }
+            """);
+
+        if (delegateIsParams)
+        {
+            compilation.VerifyDiagnostics();
+        }
+        else
+        {
+            compilation.VerifyDiagnostics(
+                // (7,23): warning CS9100: Parameter 1 has params modifier in lambda but not in target delegate type.
+                //         D d = (params x) =>
+                Diagnostic(ErrorCode.WRN_ParamsArrayInLambdaOnly, "x").WithArguments("1").WithLocation(7, 23));
+        }
+
+        var tree = compilation.SyntaxTrees.Single();
+        var root = tree.GetRoot();
+        var lambda = root.DescendantNodes().OfType<LambdaExpressionSyntax>().Single();
+
+        var semanticModel = compilation.GetSemanticModel(tree);
+        var symbol = (IMethodSymbol)semanticModel.GetSymbolInfo(lambda).Symbol!;
+
+        Assert.Equal(MethodKind.LambdaMethod, symbol.MethodKind);
+        Assert.True(symbol.Parameters is [{ Name: "x", Type: IArrayTypeSymbol { ElementType.SpecialType: SpecialType.System_Int32 }, IsParams: true }]);
+    }
+
+    [Fact]
+    public void TestParams2()
+    {
+        var compilation = CreateCompilation($$"""
+            delegate void D(params int[] x);
+
+            class C
+            {
+                void M()
+                {
+                    D d = (x) =>
+                    {
+                    };
+                }
+            }
+            """).VerifyDiagnostics();
+
+        var tree = compilation.SyntaxTrees.Single();
+        var root = tree.GetRoot();
+        var lambda = root.DescendantNodes().OfType<LambdaExpressionSyntax>().Single();
+
+        var semanticModel = compilation.GetSemanticModel(tree);
+        var symbol = (IMethodSymbol)semanticModel.GetSymbolInfo(lambda).Symbol!;
+
+        Assert.Equal(MethodKind.LambdaMethod, symbol.MethodKind);
+        Assert.True(symbol.Parameters is [{ Name: "x", Type: IArrayTypeSymbol { ElementType.SpecialType: SpecialType.System_Int32 }, IsParams: false }]);
+    }
+    [Fact]
+    public void TestIOperation()
+    {
+        var compilation = CreateCompilation("""
+            delegate void D(ref int x);
+
+            class C
+            {
+                void M()
+                {
+                    D d = (ref x) => { return; };
+                }
+            }
+            """).VerifyDiagnostics();
+
+        var tree = compilation.SyntaxTrees.Single();
+        var root = tree.GetRoot();
+        var lambda = root.DescendantNodes().OfType<LambdaExpressionSyntax>().Single();
+
+        var semanticModel = compilation.GetSemanticModel(tree);
+        var operation = (IAnonymousFunctionOperation)semanticModel.GetOperation(lambda)!;
```

Comment:
nit: I think we have some helper to verify the IOperation tree shape (in a serialized string form)